### PR TITLE
Upgraded built-in suites to cylc7.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Our built-in examples were getting out of date, this adds a validation check to the test battery.

### CODE REVIEW

File: tests/validate/63-builtin-suites.t [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+file_name = sys.argv[1]
+with open(file_name, 'r') as in_file:
+    contents = in_file.read()
+    with open(file_name + '.processed', 'w+') as out_file:
+        for msg in msgs:
+            contents = re.sub(msg, '', contents)
+        #out_file.write(re.sub(r'.*naked dummy tasks detected.*\n(\+\t.*\n)+',
+        #                      '', in_file.read()))
+        out_file.write(contents)" "$1"
+}
```
Comment (oliver-sanders) [inline tests/validate/63-builtin-suites.t]: I've had to filter out some messages for certain suites to pass. Are the "offsets are normally positive" messages necessary?

File: tests/validate/63-builtin-suites.t [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Ensure that any changes to cylc haven't broken the profile-battery command
+. $(dirname $0)/test_header
+#-------------------------------------------------------------------------------
+PATHS=('examples' 'dev/suites')      # Directories to search for suites in.
+#-------------------------------------------------------------------------------
+# Generate a list of suites.
+SUITES=
+for path in ${PATHS[@]}; do
+    SUITES+=($(find "${CYLC_DIR}/${path}" -name 'suite.rc'))
+done
```
Comment (matthewrmshin) [inline tests/validate/63-builtin-suites.t]: Can be a single command?

```bash
SUITES="$(find "${CYLC_DIR}/"{examples,dev/suites} -name 'suite.rc')"
```

File: tests/validate/63-builtin-suites.t [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+#-------------------------------------------------------------------------------
+set_test_number $((( ((${#SUITES[@]} - 1)) * 2 )))
+#-------------------------------------------------------------------------------
+# Validate suites.
+for suite in ${SUITES[@]}; do
+    suite_name=$(echo ${suite:$ABS_PATH_LENGTH} | sed 's/\//-/g')
+    TEST_NAME="${TEST_NAME_BASE}${suite_name}"
+    run_ok "${TEST_NAME}" cylc validate "${suite}" -v -v
+    filter_warnings "${TEST_NAME}.stderr"
```
Comment (matthewrmshin) [inline tests/validate/63-builtin-suites.t]: Can this just be a `sed -i`? E.g.:

```bash
sed -i '
/naked dummy tasks detected/d;
/clock-\(trigger\|expire\) offsets are normally positive/d;
' "${TEST_NAME}.stderr"
```

Comment (oliver-sanders) [inline tests/validate/63-builtin-suites.t]: `naked dummy tasks detected` messages are followed by `+ <task name>` lines which have to be filtered out as well. I couldn't workout how to perform a suitable multi-line sed deletion so went with python.

File: tests/validate/63-builtin-suites.t (hunk 2 above)
Comment (oliver-sanders) [inline tests/validate/63-builtin-suites.t]: Didn't know you could do that, nice!

**Comment by hjoliver** [comment]
@oliver-sanders - do we still need `tests/validate/04-check-examples.t` now?

**Comment by oliver-sanders** [comment]
> do we still need tests/validate/04-check-examples.t now?

Nope.

**Pull request merged**